### PR TITLE
filter-tree: Use a table layout in filter tree

### PR DESCRIPTION
**Pull request by tahini**
This allows to display many columns in the tree. Columns are sortable.

Signed-off-by: Lea <lea.carlier@polymtl.ca>
Signed-off-by: Geneviève Bastien <gbastien+lttng@versatic.net>

### CODE REVIEW

File: viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/table-row.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    };
+
+    render(): React.ReactNode | undefined {
+        if (!this.props.node) {return undefined;}
+        const children = this.renderChildren();
+
+        return (
+            <React.Fragment>
+                <tr>{this.renderRow()}</tr>
+                {children}
```
Comment (chertyb) [inline viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/table-row.tsx:87]: we can call this.renderChildren() directly

File: viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/table.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    initializeSortConfig = (): void => {
+        const config: SortConfig[] = [];
+        this.sortableColumns.forEach((column: string) => {
+            config.push({column: column, sortState: sortState.default});
+        });
+        this.props.onSortConfigChange(config);
+    };
+
+    getTableHeaderKeys = (): string[] => {
```
Comment (chertyb) [inline viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/table.tsx:41]: add TODO: remove hardcoded values when trace-server can send column headers

**Review by chertyb — CHANGES_REQUESTED** [review]
table works fine, only add TODO and I'll approve the request

**Review by chertyb — APPROVED** [review]

**Comment by bhufmann** [comment]
Just to let you know that I've started looking at the patch and review is ongoing.

**Review by bhufmann — CHANGES_REQUESTED** [review]
Thanks for adding the table. It's a nice addition to the Time Graph and the XY charts. It will be useful when implementing a statistics view.

The columns are quite undefined. They depend on the extra attributes TimeGraphEntryModel implementation of each data provider. We'll need to use the columns headers and labels for each rows, that are currently defined in the TSP, but no time graph nor xy data provider has implemented it yet. Also, the trace server and tsp-typescript-client should ignore attributes that are not in the TSP. This is outside the scope of this PR.

**Comment by tahini** [comment]
I added the todo, for the table headers and label, since changes are required in the tsp-client and trace server, it can be for later patches?

**Comment by bhufmann** [comment]
> I added the todo, for the table headers and label, since changes are required in the tsp-client and trace server, it can be for later patches?

I think so. Maybe have the trace server patch [1] that only serializes fields that are in the TSP and related tsp-client patch should be merged first. Then, the time graphs won't unnecessary columns.

[1] https://git.eclipse.org/r/c/tracecompass.incubator/org.eclipse.tracecompass.incubator/+/167256

**Comment by bhufmann** [comment]
Another question, the Legend column doesn't really make sense for time graphs, right?

**Comment by bhufmann** [comment]
> Another question, the Legend column doesn't really make sense for time graphs, right?

After testing, I realized that the Legend column is only for XY charts and not for time graph.

**Review by bhufmann — APPROVED** [review]
It looks good to me

**Comment by tahini** [comment]
Simply rebased, I'll merge for now so the update with the new tsp patches will be more obvious.

**Comment by tahini** [comment]
Thanks @chertyb for this great improvement!

**Pull request merged**